src/js/components/user-management/userlist.js

```diff
@@ -22,25 +23,34 @@ const UserList = ({ users, currentUser, editUser, removeUser }) => (
       <Table>
         <TableHead>
           <TableRow>
-            {columnData.map(column => (
-              <TableCell key={column.id} padding={column.disablePadding ? 'none' : 'default'}>
-                {column.label}
-              </TableCell>
-            ))}
+            {columnData.reduce((accu, column) => {
+              if (column.enterpriseOnly && !isEnterprise) {
+                return accu;
+              }
+              accu.push(
+                <TableCell key={column.id} padding={column.disablePadding ? 'none' : 'default'}>
+                  {column.label}
+                </TableCell>
+              );
+              return accu;
+            }, [])}
           </TableRow>
         </TableHead>
         <TableBody>
           {users.map((user, index) => (
             <TableRow key={user.id || index} hover>
               <TableCell>{user.email}</TableCell>
-              <TableCell align="left">
+              <TableCell>
                 <Time value={user.created_ts} format="YYYY-MM-DD HH:mm" />
               </TableCell>
-              <TableCell align="left">
+              <TableCell>
                 <RelativeTime updateTime={user.updated_ts} />
               </TableCell>
-              <TableCell padding="none">
-                <Button onClick={() => editUser(user)}>Edit</Button>
+              {isEnterprise && <TableCell>{(user.roles || []).map(role => roles.find(currentRole => currentRole.id === role).title).join(', ')}</TableCell>}
```

Comment:
what if
```js
currentRole => currentRole.id === role
```
is null?

Reply:
that's imported from https://github.com/mendersoftware/gui/pull/685/files#diff-3e329d83c7c7d5c0b9a0a57bca4db744R26 so unless we insert some ill defined value there, this should be defined and if the map call before is executed on an empty list so `[].map(x => x).join(', ')` the join is just ignored (so will just return `''`) 🙂

Reply:
it is all assuming that `userConstants.USER_ROLES` contains all roles that are possible. what happens when user has a role that is not defined in `USER_ROLES`? (it can happen, and we have no control over it here.)

Reply:
argh you're right! + now that you mention it, the constants alone won't be sufficient, since the group based roles e.g. would already break it... => so I'll have to get the roles from the store instead and there I have full control about the structure... 👍 thanks for spotting this I'll rework it!